tests/core/test_suite_factory.py

```diff
@@ -0,0 +1,196 @@
+    store.get.return_value = suite_dict
+    factory = SuiteFactory(store=store, include_rendered_content=False)
+    context = Mock(spec=AbstractDataContext)
+    set_context(context)
+
+    # Act
+    with pytest.raises(
+        DataContextError, match=f"ExpectationSuite with name {name} was not found."
+    ):
+        factory.get(name=name)
+
+    # Assert
+    store.get.assert_not_called()
+
+
+@pytest.mark.unit
+def test_suite_factory_add_uses_store_add():
+    # Arrange
+    name = "test-suite"
+    store = Mock(spec=ExpectationsStore)
+    store.has_key.return_value = False
+    key = store.get_key.return_value
+    factory = SuiteFactory(store=store, include_rendered_content=False)
+    context = Mock(spec=AbstractDataContext)
+    set_context(context)
+    suite = ExpectationSuite(name=name)
+
+    # Act
+    factory.add(suite=suite)
+
+    # Assert
+    store.add.assert_called_once_with(key=key, value=suite)
+
+
+@pytest.mark.unit
+def test_suite_factory_add_raises_for_duplicate_key():
+    # Arrange
+    name = "test-suite"
+    store = Mock(spec=ExpectationsStore)
+    store.has_key.return_value = True
+    factory = SuiteFactory(store=store, include_rendered_content=False)
+    context = Mock(spec=AbstractDataContext)
+    set_context(context)
+    suite = ExpectationSuite(name=name)
+
+    # Act
+    with pytest.raises(
+        DataContextError,
+        match=f"Cannot add ExpectationSuite with name {suite.name} because it already exists.",
+    ):
+        factory.add(suite=suite)
+
+    # Assert
+    store.add.assert_not_called()
+
+
+@pytest.mark.unit
+def test_suite_factory_delete_uses_store_remove_key():
+    # Arrange
+    name = "test-suite"
+    store = Mock(spec=ExpectationsStore)
+    store.has_key.return_value = True
+    key = store.get_key.return_value
+    factory = SuiteFactory(store=store, include_rendered_content=False)
+    context = Mock(spec=AbstractDataContext)
+    set_context(context)
+    suite = ExpectationSuite(name=name)
+
+    # Act
+    factory.delete(suite=suite)
+
+    # Assert
+    store.remove_key.assert_called_once_with(
+        key=key,
+    )
+
+
+@pytest.mark.unit
+def test_suite_factory_delete_raises_for_missing_suite():
+    # Arrange
+    name = "test-suite"
+    store = Mock(spec=ExpectationsStore)
+    store.has_key.return_value = False
+    factory = SuiteFactory(store=store, include_rendered_content=False)
+    context = Mock(spec=AbstractDataContext)
+    set_context(context)
+    suite = ExpectationSuite(name=name)
+
+    # Act
+    with pytest.raises(
+        DataContextError,
+        match=f"Cannot delete ExpectationSuite with name {suite.name} because it cannot be found.",
+    ):
+        factory.delete(suite=suite)
+
+    # Assert
+    store.remove_key.assert_not_called()
+
+
+@pytest.mark.filesystem
+def test_suite_factory_is_initialized_with_context_filesystem(empty_data_context):
+    assert isinstance(empty_data_context.suites, SuiteFactory)
```

Comment:
tests where the assert basically just confirms what mypy should confirm feel off. Is this just ensuring that no exception is raised?

Reply:
there is some initialization logic for the SuiteFactory, so I think this is worth having under test.

Reply:
Yeah, that's fair. My comment there was contingent on us ensuring that `_suites` was never `None`